doc/how-to-build.md

```diff
@@ -109,6 +109,49 @@ Makefile rules:
 Environment variable can be used to configure APISIX. Please take a look at `conf/config.yaml` to
 see how to do it.
 
+### Troubleshooting
+**Compile and install luasocket**
+
+If you run `make init` or `make run` to get an issue `Symbol not found: _luaL_openlib`, make sure to use the right version of luaSocket and lua. You can change lua 5.3.5 using LuaJIT or lua 5.1, if it still doesn't work, you can manually compile and install luasocket.
+
+- Download the source code to compile and install luasocket.
+
+```sh
+git clone https://github.com/diegonehab/luasocket.git
+make 
+make install 
+```
+
+If you get error as below:
+
+```
+/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src linux
+/Applications/Xcode.app/Contents/Developer/usr/bin/make all-unix PLAT=linux
+gcc  -I/usr/include/lua/5.1 -I/usr/include/lua5.1 -DLUASOCKET_NODEBUG -Wall -Wshadow -Wextra -Wimplicit -O2 -ggdb3 -fpic   -c -o luasocket.o luasocket.c
+In file included from luasocket.c:15:
+./luasocket.h:27:10: fatal error: 'lua.h' file not found
+#include "lua.h"
+         ^~~~~~~
+1 error generated.
+make[2]: *** [luasocket.o] Error 1
+make[1]: *** [linux] Error 2
+make: *** [linux] Error 2
+```
+
+Lua.h file was not found at compile time, installed in the system is lua 5.3.5_1, modify the compiler parameters:
```

Comment:
It is recommended to use LuaJIT if Lua 5.1 doesn't work.
Using Lua 5.3 may cause more problems, since we don't test it.

Reply:
Using Lua 5.3.5 is just for compile luasocket, not for the apisix

Reply:
The binary compiled with Lua 5.3 might not work properly with APISIX which runs under LuaJIT. We don't want to deal with weird problem.

Reply:
This should be the setting of OS environment variables. The default Lua version of the operating system may not be 5.1 (maxos: Lua 5.4, fedora32: Lua 5.3). 
But In APIXIX some modules or dependencies require Lua 5.1, such as luasocket.
Therefore, you need to modify the local environment to install Lua 5.1.
 If you want to continue to use Lua 5.3 or 5.4, you must use Lua 5.3 or 5.4 to recompile and install dependencies.
 I think it's better to remind users in the document : `APISIX is recommended to use LuaJIT or Lua 5.1`.

Reply:
for APISIX, we should use `luajit` or `Lua 5.1` always.

For Luarocks 2.* , it works with Lua 5.1 normally, that is fine.
For Luarocks 3.*,  it may work with Lua `5.1`, `5.2` or `5.3`. so we use `luarocks install --lua-dir=$(LUAJIT_DIR) rockspec/apisix-master-0.rockspec --tree=deps --only-deps --local` in [makefile](https://github.com/apache/apisix/blob/master/Makefile#L55) . It means we will use `Luajit`.